rfcs/text/0000-multi-data-collection-dataset-processing.md

```diff
@@ -0,0 +1,142 @@
+
+Data groups are implemented as a new bundle type that contains a JSON file
+listing the FQIDs (UUIDs with versions) of all bundles in the data group. The
+existing DSS subscription mechanism is used for notifications.
+
+A new schema *data_group* will be created that contains the fields:
+- *scope* - Symbolic name of the scope, for example, *PROJECT_SCOPE*.
+- *step* - Symbolic name of the life-cycle step, for example *hca/primary-data; hca-data-type:sptx*.
+- *bundle_fqids* - List of FQIDs of bundles that are in the group.
+
+
+
+See also:
+- [The Ingest Submission Data Model](https://docs.google.com/document/d/1qlgIROPK4qESy9-Lwvva1ZWJ3S5EdZ0HWJiGGvt2JRM/edit#heading=h.8r5egyz5ly82) for a description of submissions.
+- [Bundling in ingest](https://docs.google.com/document/d/1cGfolHMGEe1IKkXSBS75SAdSri9Np_oW6Z4S9-Ax1QY/edit#heading=h.xv0ag3c16x15) for details on how bundles are created.
+
+Initially, only the *PROJECT_SCOPE* scope will be implemented. *PROJECT_SCOPE* will indicate that a single submission of data to a project is version-complete and will be generated by Ingest only after all data bundles are committed to the DSS. The generation of the *PROJECT_SCOPE* data group indicates that data in the submission to the project should be processed by downstream components.  Ingest is responsible for creating *PROJECT_SCOPE* *data groups* after the relevant bundles have been created.  Figure 1 shows a diagram of a project with multiple *data groups*.
+
+![Figure 1](../images/0000-multi-data-collection-data-processing-images/project-submission-in-project.png)
+
+
+## Application to analysis pipelines
+
+While this RFC proposes a general mechanism, it was developed in response to the needs of analysis to co-process data that are in multiple assay bundles.  This section uses co-processing of multiple sequencing run from a given library as an example of the use of data groups to identify data that should be processed together.  A key concept behind this proposal is that upstream submission should not be defining how downstream analysis groups inputs for processing.  The analysis pipelines need to be able to group input based on ways that may not be defined at submission time.  New analysis pipelines should be able to group inputs in arbitrary ways without requiring changes to the way data is packaged.
+
+### Identifying data to co-process
+Pipelines that require grouping data from multiple sequencing assays (co-processing) subscribe to *PROJECT_SCOPES* events instead of per-bundle events. Unlike with the assay bundle per pipeline run, this makes the pipeline framework responsible for collecting assay bundles into pipeline runs.  Pipelines that do not require co-processing continue to operate on a per-bundle basis, however they still need to subscribe to the project data group so they can notify ingest when the input *data group* process is complete for use by downstream consumers, such as Azul.
```

Comment:
I think this would be more accurate if it said `the pipeline service needs to subscribe to the project data group...` 

The difference is that if every _pipeline_ subscribed to "data group notifications", whenever a project is submitted the same notification would get sent to each pipeline. This creates a lot of duplicated work in terms of inspecting the contents to determine if that pipeline should be run so we would only want a single subscription to data group notifications.

Reply:
thanks, @samanehsan text has been clarified